### ISSUE TITLE
Wall velocity for Eulerian SPH is not correct.

### ISSUE THREAD

**Issue by Bo-Zhang1995** [Xiangyu-Hu/SPHinXsys#476]
https://github.com/Xiangyu-Hu/SPHinXsys/blob/c2c30ecc842e8ac1397be7424ee92e144bf0c335/src/shared/particle_dynamics/fluid_dynamics/eulerian_fluid_dynamics/eulerian_fluid_integration.hpp#L141

The wall speed is not correct for the Eulerian SPH when it involves the wall.
For example, if the wall has a speed, like lip driven cavity, the wall speed should not be the negative speed of the particle i.
It should be like the LG sph, vel_j = 2 * vel_wall - v_i.

**Comment by Xiangyu-Hu** [comment]
@ZhentongWang could you have a look at it? Thanks.

**Comment by ZhentongWang** [comment]
> https://github.com/Xiangyu-Hu/SPHinXsys/blob/c2c30ecc842e8ac1397be7424ee92e144bf0c335/src/shared/particle_dynamics/fluid_dynamics/eulerian_fluid_dynamics/eulerian_fluid_integration.hpp#L141
> 
> The wall speed is not correct for the Eulerian SPH when it involves the wall. For example, if the wall has a speed, like lip driven cavity, the wall speed should not be the negative speed of the particle i. It should be like the LG sph, vel_j = 2 * vel_wall - v_i.

Yes, the velocity in wall as vel_in_wall = 2 * vel_ave - v_i. I was just considering the static wall boundary condition for Eulerian cases in current, so the vel_ave  is always zero.  When I simulate the lid-driven cavity in Eulerian SPH method, I also use the equation  vel_in_wall = 2 * vel_ave - v_i but have not changed that equation in the current master. Therefore, the general equation for velocity in wall boundary is vel_j = 2 * vel_wall - v_i.

**Comment by Xiangyu-Hu** [comment]
@ZhentongWang So, should we change the code on this?

**Comment by ZhentongWang** [comment]
> @ZhentongWang So, should we change the code on this?

yes, I will change the code and use the more general equation for wall velocity soon.

**Comment by Xiangyu-Hu** [comment]
> > @ZhentongWang So, should we change the code on this?
> 
> yes, I will change the code and use the more general equation for wall velocity soon.



> > @ZhentongWang So, should we change the code on this?
> 
> yes, I will change the code and use the more general equation for wall velocity soon.



> > @ZhentongWang So, should we change the code on this?
> 
> yes, I will change the code and use the more general equation for wall velocity soon.

@ZhentongWang Is this done already?

**Comment by ZhentongWang** [comment]
> > > @ZhentongWang So, should we change the code on this?
> > 
> > 
> > yes, I will change the code and use the more general equation for wall velocity soon.
> 
> > > @ZhentongWang So, should we change the code on this?
> > 
> > 
> > yes, I will change the code and use the more general equation for wall velocity soon.
> 
> > > @ZhentongWang So, should we change the code on this?
> > 
> > 
> > yes, I will change the code and use the more general equation for wall velocity soon.
> 
> @ZhentongWang Is this done already?

I have created a pull request to have it fixed.
 i.e. Vecd vel_in_wall = 2.0 * vel_ave_k[index_j] - state_i.vel_.

**Comment by Xiangyu-Hu** [comment]
@ZhentongWang  Could link this issue to the branch for fixing?
